IO/src/main/java/io/deephaven/io/streams/ByteBufferOutputStream.java

```diff
@@ -128,65 +124,49 @@ public void write(ByteBuffer b) throws IOException {
 
     @Override
     public void writeBoolean(boolean v) throws IOException {
-        if (buf.remaining() < 1) {
-            buf = sink.acceptBuffer(buf, 1);
-        }
+        ensureRemaining(Byte.BYTES);
         buf.put((byte) (v ? 1 : 0));
     }
 
     @Override
     public void writeByte(int v) throws IOException {
-        if (buf.remaining() < 1) {
-            buf = sink.acceptBuffer(buf, 1);
-        }
+        ensureRemaining(Byte.BYTES);
         buf.put((byte) v);
     }
 
     @Override
     public void writeShort(int v) throws IOException {
-        if (buf.remaining() < 2) {
-            buf = sink.acceptBuffer(buf, 2);
-        }
+        ensureRemaining(Short.BYTES);
         buf.putShort((short) v);
     }
 
     @Override
     public void writeChar(int v) throws IOException {
-        if (buf.remaining() < 2) {
-            buf = sink.acceptBuffer(buf, 2);
-        }
+        ensureRemaining(Character.BYTES);
         buf.putChar((char) v);
     }
 
     @Override
     public void writeInt(int v) throws IOException {
-        if (buf.remaining() < 4) {
-            buf = sink.acceptBuffer(buf, 4);
-        }
+        ensureRemaining(Integer.BYTES);
         buf.putInt(v);
     }
 
     @Override
     public void writeLong(long v) throws IOException {
-        if (buf.remaining() < 8) {
-            buf = sink.acceptBuffer(buf, 8);
-        }
+        ensureRemaining(Long.BYTES);
         buf.putLong(v);
     }
 
     @Override
     public void writeFloat(float f) throws IOException {
-        if (buf.remaining() < 8) {
-            buf = sink.acceptBuffer(buf, 8);
-        }
```

Comment:
While this code works, it's a bit sloppier than using 4 (Float.BYTES).

Reply:
Why would we want to reserve 8 bytes for a 4 byte thing? Seems to me like your version is just plain better.

Reply:
Yes, just noting that technically the logic has changed here.